src/protocol/http/src/main/java/org/apache/jmeter/protocol/http/correlation/CreateRegexExtractor.java

```diff
@@ -0,0 +1,195 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to you under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.jmeter.protocol.http.correlation;
```

Comment:
Aren't those XXXExtractor classes good candidate for Command Pattern and base class + interface ?

Reply:
Migrated the Extractor creator classes to command design pattern.

Reply:
Thank you for the suggestion. We have added Command Pattern in the CreateExtractorInterface and also
made them inherit from the base class.
The following new classes were added to make this change possible:
	CreateBoundaryExtractor
	CreateCssSelectorExtractor
	CreateJsonPathExtractor
	CreateRegexExtractor
	CreateXPath2Extractor

Reply:
Thank you for the suggestion. We have added Command Pattern in the CreateExtractorInterface and also made them inherit from the base class.
The following classes modified  to make this change possible:
	CreateBoundaryExtractor
	CreateCssSelectorExtractor
	CreateJsonPathExtractor
	CreateRegexExtractor
	CreateXPath2Extractor